[Lim Wei Chang] iP

weichang18 opened the pull request:
### Duke is program that helps you record down your task similarly to writing in your diary. It's,

-  text-based
- easy to use and lean
- convenient

You just need to:

1. download it [here](https://github.com/weichang18/ip/releases/tag/A-Jar)
2. open it
3. input your task
4. and its saved

Best of all its **FREE**!😉

Features:

- [x]  Managing tasks

- [ ]  Managing deadlines (coming soon)

- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Duke {
    public static void `main`(String[] args) {
        (new Duke("data/dukeTaskList.txt")).run();
    }
}
```

As David Allen said:
>Your mind is for having ideas, not holding them.

Commit:
Add Gradle support

weichang18 committed:
Level-1

weichang18 committed:
level-2

weichang18 committed:
level-3

weichang18 committed:
Level-4

weichang18 committed:
A-TextUiTesting

weichang18 committed:
Level-5

weichang18 committed:
no message

weichang18 committed:
Level-6

weichang18 committed:
Add saving onto hardisk capabilities

weichang18 committed:
Add date and time

weichang18 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

weichang18 committed:
Add date and time

weichang18 committed:
Resolve bugs

weichang18 committed:
Resolve bugs

weichang18 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

weichang18 committed:
Add Ui, Storage, Parser, TaskList classes

weichang18 committed:
Put all classes into one package

weichang18 committed:
Add JUnit tests for Deadline.java and Parser.java

weichang18 committed:
Add .gitignore and .xml files

weichang18 committed:
Add ignore files

weichang18 committed:
Package file into JAR file

weichang18 committed:
Add JavaDoc and tweak code to comply with coding standard

weichang18 committed:
Adjusted JavaDoc for exception classes

weichang18 committed:
Addtional tweaks to coding standard

weichang18 committed:
Add ability to find task with keyword

weichang18 committed:
Merge branch 'branch-A-CodingStandard'

weichang18 committed:
Merge branch 'branch-Level-9'

weichang18 committed:
Merge branch 'add-gradle-support'

weichang18 committed:
Add checksyle on gradle and made some tweaks to coding standard and made some edits to Task.java and TaskList.java

luminousleek commented on .gitignore:
```diff
@@ -15,3 +15,32 @@ bin/
 
 /text-ui-test/ACTUAL.txt
 text-ui-test/EXPECTED-UNIX.TXT
+src/test/out/production/main/duke/Deadline.class
```
Not a code quality issue, but instead of having to manually add all the class files and whatnot into your `.gitignore` file, maybe can just do this:
```
/src/test/out/
/src/main/out/
```
and it should work, unless you have other stuff in the out subfolders that you want to get stored on Git. Then I guess you could do
```
src/test/out/production/main/duke/*.class
src/test/out/test/test/duke/*.class
src/main/out/production/main/duke/*.class
```

luminousleek replied:
Might also want to add `/.idea/` to your `.gitignore`?

luminousleek commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,122 @@
+                        } else {
+                            newTask = new Deadline(parser.deadlineInputTaskParser(input),
+                                    parser.deadlineInputDateParser(input));
+                        }
```
Quite deep nesting here - maybe you can put the block of code highlighted here into a new method called `deadlineInputParser`? Same thing with the event input below

luminousleek commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,122 @@
+                    //Initialise the task if its a valid input.
+                    Task newTask = null;
+                    if (input.contains("todo")) {
+                        newTask = new Todo(parser.toDoInputParser(input));
```
You might want to consider making the various task parsers return a task? Might make your life easier.

luminousleek commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+    public String deadlineInputDateParser(String input) {
+        String[] parts = input.split(" /by ");
+        return parts[1];
+    }
```
Wondering if you could combine both these methods together and return a String[], i.e.
```java
public String[] deadlineInputParser(String input) {
    String[] parts = input.split( " /by ");
    parts[0] = parts[0].replace("deadline ", "");
    return parts;
}
```
Similar thing for the event parser below

luminousleek commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,103 @@
+
+public class Storage {
+    private File taskListFile;
+    private Boolean doesFileExists;
```
Sorry to be a grammar Nazi, but I think it should be named `doesFileExist` or simply `fileExists`?

luminousleek commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    public String checkIsDone() {
+        return this.isDoneBox;
+    }
```
Does the method name sound like it would return `isDone` instead of `isDoneBox`? Maybe consider renaming it to `getIsDoneBox`?

luminousleek commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     */
+    public Boolean getDoesFileExists() {
+        return this.doesFileExists;
+    }
```
Is this method used anywhere else?

luminousleek commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private int noOfTask = 0;
```
Consider changing to `numOfTask` or `numTasks`? The abbreviation No can be a bit confusing

luminousleek commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+    }
+    public int getNoOfTask() {
+        return this.noOfTask;
+    }
```
What's the difference between this method and the `taskListSize` method?

luminousleek commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,103 @@
+                    + taskList.getTask(j).checkIsDone()
+                    + " " + taskList.getTask(j).getDescription();
+            listBuilder.append(listItem).append(System.lineSeparator());
+        }
```
This seems like it would have the same output as `taskList.toString()`? Maybe can call that instead? Then change the `taskList.toString` method to use a `StringBuilder` instead.

luminousleek commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+                    + "."
+                    + taskList.get(j).getTaskType()
+                    + taskList.get(j).checkIsDone()
+                    + " " + taskList.get(j).getDescription() + "\n";
```
Maybe implement a `Task.toString()` method so you won't have to do this?

luminousleek commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,102 @@
+                + "Roger! I will add this task in: \n"
+                + task.getTaskType()
+                + task.checkIsDone()
+                + " "
```
Here you inserted a space in between the checkbox and the description, but in `deleteTaskMsg` you didn't?

luminousleek commented on src/main/java/duke/WrongInputException.java:
```diff
@@ -0,0 +1,18 @@
+
+public class WrongInputException extends DukeException {
+
+    WrongInputException() {
```
No access modifier?

luminousleek reviewed (commented):
Good job on doing up the iP till this far! The only general comment I have is to try to avoid duplicated code - if you find yourself writing the same lines of code again you should probably wrap it in a method.

All the best with this week's tasks - do get started on JavaFX early because it'll take a while to do 😣

jiarong15 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,122 @@
+    public void run() throws IOException {
+        this.isRunning = true;
+        this.ui.greetingMsg();
+        Scanner sc = new Scanner(System.in);
```
Maybe you can consider abstracting these reading of inputs into your UI class?

jiarong15 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,122 @@
+                            throw new MissingDateException();
+                        } else {
+                            newTask = new Deadline(parser.deadlineInputTaskParser(input),
+                                    parser.deadlineInputDateParser(input));
```
Agree that the method here is really long, abstract into a private method would be better 😄

jiarong15 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,50 @@
+     * @param deadline Deadline of event task in "yyyy-mm-dd HHmm" format.
+     */
+    public Event(String description, String deadline) {
+        super(description, "[E]");
```
maybe for the "[E]" can make a private static final String attribute ?

jiarong15 reviewed (commented):
Generally okay just minor changes :)

weichang18 committed:
Add GUI interface and remove text UI for duke

weichang18 committed:
Merge branch 'branch-Level-10'

weichang18 committed:
Generate a new JAR file

weichang18 committed:
Improve code quality and make changes commented by peer review

weichang18 committed:
Add assertion checks in Duke.java and TaskList.java

weichang18 committed:
Merge pull request #1 from weichang18/branch-A-CodeQuality

Improve code quality and make changes commented by peer review

weichang18 committed:
Merge branch 'master' of https://github.com/weichang18/ip

weichang18 committed:
Merge branch 'master' into branch-A-Assertions

weichang18 committed:
Merge pull request #2 from weichang18/branch-A-Assertions

Add assertion checks in Duke.java and TaskList.java

weichang18 committed:
Merge branch 'master' of https://github.com/weichang18/ip

weichang18 committed:
Add help function to show all the available commands and their format

weichang18 committed:
Inputting done or delete with an index more than task list cause error.

There is a possibility of user input index that is more than the size of
the task list.

Add Index out of bounds exception.

To catch the exception and prints and error message if user input
the wrong index.

weichang18 committed:
Parser does not check find command contains keyword.

Causes error if it does not contain a keyword.

Add new missing keyword exception.

To catch and prints error message if user forgot to add keyword.

weichang18 committed:
Add new class MissingIndexException
Adjusted help function to return a more readable message

weichang18 committed:
Fix bugs in Parser and Ui class and make small adjustments to GUI

weichang18 committed:
Add screenshot of PokeBot
Add user guide
Adjusted GUI to look more attractive

weichang18 committed:
Set theme jekyll-theme-cayman

weichang18 committed:
Merge branch 'master' of https://github.com/weichang18/ip

weichang18 committed:
Make changes to build gradle for v0.2 release
